src/argilla/server/contexts/datasets.py

```diff
@@ -788,6 +788,19 @@ async def _update_record(
     return params, suggestions, vectors, needs_search_engine_update, caches
 
 
+async def refresh_records(records: List[Record]):
+    # TODO: This behavior must be reviewed !!!
+    for record in records:
+        await record.awaitable_attrs.suggestions
+        await record.awaitable_attrs.responses
+        await record.awaitable_attrs.vectors
+
+        for response in record.responses:
+            await response.awiatable_attrs.user
+        for suggestion in record.suggestions:
+            await suggestion.awaitable_attrs.question
```

Comment:
I think we can improve the naming in the function and separate it having one for the collection and one for a single item:

```suggestion
async def _preload_records_associations(records: List[Record]) -> None:
    for record in records:
        await _preload_record_associations(record)


async def _preload_record_associations(record: Record) -> None:
    # TODO: This behavior must be reviewed !!!
    await record.awaitable_attrs.suggestions
    await record.awaitable_attrs.responses
    await record.awaitable_attrs.vectors

    for response in record.responses:
        await response.awiatable_attrs.user

    for suggestion in record.suggestions:
        await suggestion.awaitable_attrs.question


```